### PR TITLE
Add `_print_name` to Truncated and CustomDists

### PR DESCRIPTION
**Pull request by ricardoV94**
## Checklist

- [x] Checked that [the pre-commit linting/style checks pass](https://docs.pymc.io/en/latest/contributing/python_style.html)
- [x] Included tests that prove the fix is effective or that the new feature works
- [x] Added necessary documentation (docstrings and/or example notebooks)
- [x] If you are a pro: each commit corresponds to a [relevant logical change](https://wiki.openstack.org/wiki/GitCommitMessages#Structural_split_of_changes)

## Type of change

- [x] New feature / enhancement
- [ ] Bug fix
- [ ] Documentation
- [ ] Maintenance
- [ ] Other (please specify):

----
📚 Documentation preview 📚: https://pymc--7205.org.readthedocs.build/en/7205/

### CODE REVIEW

**Comment by jessegrabowski** [comment]
Looks good. Is Truncated the only "wrapper" distribution that needs special treatment? I thinking about`Censored`, maybe?

**Review by jessegrabowski — APPROVED** [review]

**Comment by ricardoV94** [comment]
> Looks good. Is Truncated the only "wrapper" distribution that needs special treatment? I thinking about`Censored`, maybe?

Truncated is special because it doesn't just operate on the input dist, but does either icdf or rejection sampling, so it doesn't print the name nicely, whereas for Censored you see something like `Censored(normal(0, 1), ...)`

File: tests/test_printing.py [hunk 1]
```diff
@@ -285,3 +285,27 @@ def test_model_repr_variables_without_monkey_patched_repr():
...
+    with Model() as model:
+
+        def dist(mu, size):
+            return Normal.dist(mu, 1, size=size)
+
+        def random(rng, mu, size):
+            return rng.normal(mu, size=size)
+
+        x = CustomDist("x", 0, dist=dist, class_name="CustomDistNormal")
+        x = CustomDist("y", 0, random=random, class_name="CustomRandomNormal")
```
Comment (larryshamalama) [inline tests/test_printing.py:308]:
``` suggestion
        y = CustomDist("y", 0, random=random, class_name="CustomRandomNormal")
```
typo?

Comment (ricardoV94) [reply tests/test_printing.py:308]: Yeah, it's not used anywhere though

**Pull request merged**